retrieval/targetmanager_test.go

```diff
@@ -12,3 +12,63 @@
 // limitations under the License.
 
 package retrieval
+
+import (
+	"testing"
+
+	"golang.org/x/net/context"
+	"gopkg.in/yaml.v2"
+
+	"github.com/prometheus/prometheus/config"
+	"github.com/prometheus/prometheus/storage/local"
+)
+
+func TestTargetSetRecreatesTargetGroupsEveryRun(t *testing.T) {
+	scrapeConfig := &config.ScrapeConfig{}
+
+	sOne := `
+job_name: "foo"
+dns_sd_configs:
+- names:
+  - "srv.name.one.example.org"
+`
+	if err := yaml.Unmarshal([]byte(sOne), scrapeConfig); err != nil {
+		t.Fatalf("Unable to load YAML config sOne: %s", err)
+	}
```

Comment:
Let's generate the output of `providersFromConfig` directly and don't go the indirection via the function + YAML.